tools/check-readme/main.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"slices"
+	"strings"
+)
+
+func do() error {
+	// Read all lines from the kubectl-gadget --help output
+	var outb, errb bytes.Buffer
+	cmd := exec.Command(os.Args[1], "--help")
+	cmd.Stderr = &errb
+	cmd.Stdout = &outb
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("running: %w. stderr: %s", err, errb.String())
+	}
+
+	kubectlGadgetLines := strings.Split(outb.String(), "\n")
+
+	file, err := os.ReadFile(os.Args[2])
+	if err != nil {
+		return err
+	}
+
+	// Read the whole readme and find where the $ kubectl gadget --help part is
+	readmeLines := strings.Split(string(file), "\n")
+	startIndex := slices.Index(readmeLines, "$ kubectl gadget --help")
+	if startIndex == -1 {
+		return fmt.Errorf("finding '$ kubectl gadget --help'")
+	}
+	endIndex := slices.Index(readmeLines[startIndex:], "...")
+	if endIndex == -1 {
+		return fmt.Errorf("finding '...'")
+	}
+
+	// Remove unuseful things
+	readmeLines = readmeLines[startIndex+1 : startIndex+endIndex]
+	kubectlGadgetLines = kubectlGadgetLines[:len(readmeLines)-1]
+
+	// Compare line by line. TODO: compare whole thing and print preffy diff
+	for i := 0; i < len(kubectlGadgetLines); i++ {
+		if kubectlGadgetLines[i] != readmeLines[i] {
```

Comment:
Check `len(readmeLines)` before the for loop to ensure that `readmeLines[i]` does not panic.